api/src/main/java/io/druid/data/input/FirehoseFactoryV2.java

```diff
@@ -41,5 +39,5 @@
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
 public interface FirehoseFactoryV2<T extends InputRowParser>
 {
-  FirehoseV2 connect(T parser, Object lastCommit) throws IOException, ParseException;
+  FirehoseV2 connect(T parser, Object lastCommit) throws ParseException;
```

Comment:
am wondering why we are removing IOException, seems reasonable that such method will throw IOException?

Reply:
Because nobody actually throwing it

Reply:
But it is legitimate kind of exception that maybe in the future or other external implementations will need it, `FirehoseFactoryV2.java` is an extension point, IMO we should keep it.

Reply:
Reverted